capnp/compile.lua

```diff
@@ -524,7 +524,8 @@ end
 function comp_serialize(res, name)
     insert(res, format([[
 
-    serialize = function(data, p8, size)
+    -- Serialize and return pointer to char[] and size
+    serialize_cdata = function(data, p8, size)
         if not p8 then
```

Comment:
this should be `if p8 == nil then...` remember that  `not ffi.cast('uint8_t*', nil)` is `true`

Reply:
Yes, that's better